swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala

```diff
@@ -157,6 +157,27 @@ private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
     linearizeStack(rr.path::Nil).flatMap(go(_, None)).headOption
   }
 
+  def collectSecurityScopes(rr: RhoRoute[_]): List[Map[String, List[String]]] = {
+
+    def go(stack: List[PathOperation], summary: Option[Map[String, List[String]]]): Option[Map[String, List[String]]] =
+      stack match {
+        case PathMatch("")::Nil                => go(Nil, summary)
+        case PathMatch(s)::xs                  => go(xs, summary)
+        case PathCapture(id, _, parser, _)::xs => go(xs, summary)
+        case CaptureTail::xs                   => summary
+
+        case MetaCons(_, meta)::xs =>
+          meta match {
+            case RouteSecurityScope(secScope) => secScope.some
```

Comment:
From the look of the function it seems that we potentially want to continue searching until the end, accumulating any data, otherwise we can eliminate the `summary` parameter.

Reply:
Also good point. I reduced unnecessary checks and removed summary.